src/apps/pages/RecordDetail/GeoHeader.astro

```diff
@@ -0,0 +1,58 @@
+---
+import UserDefined from '@apps/pages/RecordDetail/UserDefined.astro';
+import Identifiers from '@apps/pages/RecordDetail/Identifiers.astro';
+import PlaceMap from '@apps/pages/RecordDetail/PlaceMap';
+import RelatedPlaces from '@apps/pages/RecordDetail/RelatedPlaces.astro';
+import ServiceFactory from '@services/coreData/factory';
+import { getCoverImage, getRelatedGeometry } from '@utils/detailPages';
+
+interface Props {
+  excludes: string[];
+  lang: string;
+  name: string;
+  record: any;
+  model: string
+}
+
+const { excludes, lang, record, model, name } = Astro.props;
+
+const service = ServiceFactory.getService(model);
+const { places } = await service.getRelatedPlaces(record.uuid, true);
+const { mediaContents } = await service.getRelatedMediaContents(record.uuid);
+
+const geometry = model === 'places'
```

Comment:
I'm not sure we want to be checking `model === <value>` in these child components, I think it could end up getting very messy. Could we instead include that business logic as a prop that is passed to this component?

Reply:
We can't pass functions as props to `server:defer` components because the props are serialized and sent to the server, so any `onLoad` or `getGeometry` type of thing isn't possible (this is also why we're looking up the service via the factory multiple times).

Reply:
For the place model, could we provide the geometry as a prop? Then do some sort of logic like: 

`geometry || getRelatedGeometry(places)`?

This way the places page is still responsible for the place-specific business logic.

Reply:
I think fetching the related places within the `GeoHeader` component is fine. I'm suggesting we move the `model === 'place'` specific logic _outside_ of this component  onto the place page, and pass the required data as a prop.